### PR TITLE
Upgrade to net core 2.0

### PR DESCRIPTION
**Pull request by kburtram**

### CODE REVIEW

**Comment by kburtram** [comment]
I expect the CI build to fail initially.  I'll likely need to go through a few iterations to get those environments building clean.

File: build.cake [hunk 1]
```diff
@@ -195,10 +195,10 @@ Task("Restore")
     .IsDependentOn("Setup")
     .Does(() =>
 {
-    RunRestore(dotnetcli, "restore", sourceFolder)
+    RunRestore(dotnetcli, "restore", workingDirectory)
         .ExceptionOnError("Failed to restore projects under source code folder.");
-    RunRestore(dotnetcli, "restore --infer-runtimes", testFolder)
-        .ExceptionOnError("Failed to restore projects under test code folder.");
+    // RunRestore(dotnetcli, "restore --infer-runtimes", testFolder)
```
Comment (kburtram) [inline build.cake]: remove dead code.

File: build.cake [hunk 2]
```diff
@@ -250,9 +250,9 @@ Task("TestCore")
 
     foreach (var testProject in testProjects)
     {
-        var logFile = System.IO.Path.Combine(logFolder, $"{testProject}-core-result.xml");
+        var logFile = System.IO.Path.Combine(logFolder, $"{testProject}-core-result.trx");
```
Comment (kburtram) [inline build.cake]: dotnet no longer supports xml output, so we need to update CI builds to read TFX files

File: src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoQuerier.cs [hunk 3]
```diff
@@ -58,9 +58,9 @@ protected IDataReader GetDataReader(object data)
                 reader = ((Data.Tools.DataSets.DataTable)data).CreateDataReader();
             }
            
-            else if (data is DataSet)
+            else if (data is Microsoft.Data.Tools.DataSets.DataSet)
```
Comment (kburtram) [inline src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoQuerier.cs]: I'll remove private "DataSets" dependency in a separate PR after getting initial conversion done.

File: build.cake (hunk 2 above)
Comment (benrr101) [inline build.cake]: T*F*X or T*R*X?

File: global.json [hunk 4]
```diff
@@ -1,8 +1 @@
-{
-  "projects": [ "src", "test" ],
-  "sdk": {
-    "version": "1.0.0-preview2-003156"
-  }
-} 
-
-
+{"projects":["src","test"]}
```
Comment (benrr101) [inline global.json]: Yay! No more hard-coded sdk versions!

**Comment by llali** [comment]
:shipit:

**Comment by kburtram** [comment]
@benrr101 I've opened this issue to track the disabled unit tests https://github.com/Microsoft/carbon/issues/914.  I'll try to re-enable before merging, but that may be a follow-up PR.  

Same note on the program.cs files.  Currently dotnet 2.0 generates an default entry point for projects containing unit tests which conflicts with the explicit entry point we define.  I'll see if there is a workaround before merging, but this may happen post initial merge.

File: build.cake (hunk 2 above)
Comment (kburtram) [inline build.cake]: TRX is the file extension for mstest output, which is now the dotnet test output format.

**Comment by benrr101** [comment]
@kburtram is this version of the PR ready for more review?

**Comment by kburtram** [comment]
AppVeyor and Coveralls are passing so I'm merging.  Please ping me for anything that breaks from this merge.

**Pull request merged**